marker_rustc_driver/src/conversion/marker/stmts.rs

```diff
@@ -1,37 +1,44 @@
-use marker_api::{
-    ast::stmt::{LetStmt, StmtKind},
-    CtorBlocker,
-};
+use marker_api::ast::stmt::{CommonStmtData, ExprStmt, ItemStmt, LetStmt, StmtKind};
 use rustc_hir as hir;
 
 use super::MarkerConverterInner;
 
 impl<'ast, 'tcx> MarkerConverterInner<'ast, 'tcx> {
+    #[must_use]
     pub fn to_stmt(&self, stmt: &hir::Stmt<'tcx>) -> Option<StmtKind<'ast>> {
-        match &stmt.kind {
+        let id = self.to_stmt_id(stmt.hir_id);
+        if let Some(stmt) = self.stmts.borrow().get(&id) {
+            return Some(*stmt);
+        }
+
+        let data = CommonStmtData::new(self.to_stmt_id(stmt.hir_id), self.to_span_id(stmt.span));
+        let stmt = match &stmt.kind {
             hir::StmtKind::Local(local) => match local.source {
-                hir::LocalSource::Normal => Some(StmtKind::Let(self.alloc(LetStmt::new(
-                    self.to_let_stmt_id(local.hir_id),
-                    self.to_span_id(local.span),
+                hir::LocalSource::Normal => StmtKind::Let(self.alloc(LetStmt::new(
+                    data,
                     self.to_pat(local.pat),
                     local.ty.map(|ty| self.to_syn_ty(ty)),
                     local.init.map(|init| self.to_expr(init)),
                     local.els.map(|els| self.to_expr_from_block(els)),
-                )))),
+                ))),
                 hir::LocalSource::AssignDesugar(_) => {
                     unreachable!("this will be handled by the block expr wrapping the desugar")
                 },
                 hir::LocalSource::AsyncFn | hir::LocalSource::AwaitDesugar => {
                     eprintln!("skipping not implemented statement at: {:?}", stmt.span);
-                    None
+                    None?
```

Comment:
That's quite a hacky way to just say this:
```suggestion
                    return None;
```
The problem with `None?` is that `rustc`'s code flow analysis doesn't infer that this line of code returns, and thus won't infer an uninhabited (`!`) type of the expression here which may be important in some cases.